library/src/main/java/com/google/android/cameraview/CameraView.java

```diff
@@ -95,14 +96,15 @@ public CameraView(Context context, AttributeSet attrs, int defStyleAttr) {
             return;
         }
         // Internal setup
-        final PreviewImpl preview = createPreviewImpl(context);
-        mCallbacks = new CallbackBridge();
-        if (Build.VERSION.SDK_INT < 21) {
+        PreviewImpl preview = createPreviewImpl(context);
+        mCallbacks = new CallbackBridge(this);
+        final Context applicationContext = context.getApplicationContext();
+        if (SDK_INT < 21) {
             mImpl = new Camera1(mCallbacks, preview);
-        } else if (Build.VERSION.SDK_INT < 23) {
-            mImpl = new Camera2(mCallbacks, preview, context);
+        } else if (SDK_INT < 23) {
+            mImpl = new Camera2(mCallbacks, preview, applicationContext);
```

Comment:
Very important: pass the application context (not the activity context) to the constructors here because it uses that context to get the camera service/manager as a system service.  There is a bug in android where the CameraManager maintains a reference to the context.  So, if you start the activity 100 times, you could have 100 of these activities taking up memory.